contest/contest_info.py

```diff
@@ -97,10 +99,13 @@ def get_scoreboard(contest):
 
     return scoreboard
 
-def get_clarifications(contest,user):
-    reply_all = Clarification.objects.filter(contest = contest, reply_all = True)
-    user_ask = Clarification.objects.filter(contest = contest, asker = user)
-    return reply_all | user_ask
+def get_clarifications(user,contest):
+    if has_contest_ownership(user,contest):
+        return Clarification.objects.filter(contest = contest)
+    else:
+        reply_all = Clarification.objects.filter(contest = contest, reply_all = True)
+        user_ask = Clarification.objects.filter(contest = contest, asker = user)
+        return reply_all | user_ask
 
 def is_contestant(user,contest):
```

Comment:
If the main purpose is to avoid query db, then i can check it before querying db.
Since this function is used in template, anonymous user can't call this func. But actucally anonymous user can still view some part of the website.